matrix/spiral_print.py

```diff
@@ -123,7 +123,9 @@ def spiral_traversal(matrix: list[list[int]]) -> list[int]:
     Stage 6.
     [1, 2, 3, 4, 8, 12, 11, 10, 9, 5, 6, 7] + spiral_traversal([])
     """
-    return matrix and list(matrix.pop(0)) + spiral_traversal(list(zip(*matrix))[::-1])
+    if matrix:
+        spiral_traverse_mat = list(matrix.pop(0)) + spiral_traversal(list(zip(*matrix))[::-1])
+        return spiral_traverse_mat
```

Comment:
```suggestion
    matrix = [[]]
    return list(matrix.pop(0)) + spiral_traversal(list(zip(*matrix))[::-1])
```

Reply:
Ohh, from long time I was trying to resolve this only,

Thanks (^_^);